### PR TITLE
refactor: move some random methods to helpers

### PR DESCRIPTION
**Pull request by Shinigami92**
After we discussed stuff in https://github.com/faker-js/faker/pull/874#issuecomment-1110198059 I move these into helpers

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -742,4 +742,62 @@ export class Helpers {
...
+   * @param array Array to pick the value from.
+   *
+   * @example
+   * faker.helpers.arrayElement(['cat', 'dog', 'mouse']) // 'dog'
+   */
+  arrayElement<T = string>(array: ReadonlyArray<T>): T {
+    const index =
+      array.length > 1
+        ? this.faker.datatype.number({ max: array.length - 1 })
+        : 0;
```
Comment (Shinigami92) [inline src/helpers.ts:759]: I'm not sure if this optimization is worth it, or if we should just generate also an index of 0 and trigger the seed sequence by that :thinking:

Comment (ST-DDT) [reply src/helpers.ts:763]: IMO it adds unnecessary complexity for a very negligible performance gain.

Comment (Shinigami92) [reply src/helpers.ts:763]: nevermind

File: src/helpers.ts [hunk 2]
```diff
@@ -742,4 +742,71 @@ export class Helpers {
...
+   *
+   * @template T The type of the entries to pick from.
+   * @param array Array to pick the value from.
+   *
+   * @example
+   * faker.helpers.arrayElement(['cat', 'dog', 'mouse']) // 'dog'
+   */
+  arrayElement<T = string>(
+    // TODO @Shinigami92 2022-04-30: We want to remove this default value, but currently it's not possible because some definitions could be empty
+    // See https://github.com/faker-js/faker/issues/***
```
Comment (Shinigami92) [inline src/helpers.ts:757]: missing issue number

File: src/helpers.ts [hunk 3]
```diff
@@ -742,4 +742,71 @@ export class Helpers {
...
+   *    When not provided, random number of elements will be picked.
+   *    When value exceeds array boundaries, it will be limited to stay inside.
+   *
+   * @example
+   * faker.helpers.arrayElements(['cat', 'dog', 'mouse']) // ['mouse', 'cat']
+   * faker.helpers.arrayElements([1, 2, 3, 4, 5], 2) // [4, 2]
+   */
+  arrayElements<T>(
+    // TODO @Shinigami92 2022-04-30: We want to remove this default value, but currently it's not possible because some definitions could be empty
+    // See https://github.com/faker-js/faker/issues/***
```
Comment (Shinigami92) [inline src/helpers.ts:783]: missing issue number

File: src/helpers.ts [hunk 4]
```diff
@@ -686,7 +686,7 @@ export class Helpers {
       name: [this.faker.finance.accountName(), this.faker.finance.mask()].join(
         ' '
       ),
-      type: this.faker.random.arrayElement(
+      type: this.faker.helpers.arrayElement(
```
Comment (ST-DDT) [inline src/helpers.ts:689]:
```suggestion
      type: this.arrayElement(
```

File: src/helpers.ts [hunk 5]
```diff
@@ -727,7 +727,7 @@ export class Helpers {
    */
   objectKey<T extends Record<string, unknown>>(object: T): keyof T {
     const array: Array<keyof T> = Object.keys(object);
-    return this.faker.random.arrayElement(array);
+    return this.faker.helpers.arrayElement(array);
```
Comment (ST-DDT) [inline src/helpers.ts:730]:
```suggestion
    return this.arrayElement(array);
```

File: src/helpers.ts [hunk 6]
```diff
@@ -249,10 +249,10 @@ export class Helpers {
       if (string.charAt(i) === '#') {
         str += this.faker.datatype.number(9);
       } else if (string.charAt(i) === '?') {
-        str += this.faker.random.arrayElement(alpha);
+        str += this.faker.helpers.arrayElement(alpha);
       } else if (string.charAt(i) === '*') {
         str += this.faker.datatype.boolean()
-          ? this.faker.random.arrayElement(alpha)
+          ? this.faker.helpers.arrayElement(alpha)
```
Comment (ST-DDT) [inline src/helpers.ts:255]:
```suggestion
          ? this.arrayElement(alpha)
```

File: src/helpers.ts [hunk 7]
```diff
@@ -249,10 +249,10 @@ export class Helpers {
       if (string.charAt(i) === '#') {
         str += this.faker.datatype.number(9);
       } else if (string.charAt(i) === '?') {
-        str += this.faker.random.arrayElement(alpha);
+        str += this.faker.helpers.arrayElement(alpha);
```
Comment (ST-DDT) [inline src/helpers.ts:252]:
```suggestion
        str += this.arrayElement(alpha);
```

File: src/helpers.ts [hunk 8]
```diff
@@ -147,11 +147,11 @@ export class Helpers {
   ): T {
     deprecated({
       deprecated: 'faker.helpers.randomize()',
-      proposed: 'faker.random.arrayElement()',
+      proposed: 'faker.helpers.arrayElement()',
       // since: 'v5.0.0', (?)
       until: 'v7.0.0',
     });
-    return this.faker.random.arrayElement(array);
+    return this.faker.helpers.arrayElement(array);
```
Comment (ST-DDT) [inline src/helpers.ts:154]:
```suggestion
    return this.arrayElement(array);
```

**Review by ST-DDT — CHANGES_REQUESTED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Pull request merged**